src/otaku.cpp

```diff
@@ -0,0 +1,197 @@
+#include <stdlib.h>
+#include <time.h>
+
+#include "otaku.hpp"
+#include "structs.hpp"
+#include "awake.hpp"
+#include "db.hpp"
+#include "comm.hpp"
+#include "interpreter.hpp"
+#include "handler.hpp"
+#include "utils.hpp"
+#include "screen.hpp"
+#include "constants.hpp"
+#include "olc.hpp"
+#include "config.hpp"
+#include "ignore_system.hpp"
+#include "newdb.hpp"
+#include "playerdoc.hpp"
+#include "newhouse.hpp"
+#include "quest.hpp"
+
+extern struct otaku_echo echoes[];
+
+long calculate_sub_nuyen_cost(int desired_grade) {
+  if (desired_grade >= 6)
+    return 825000;
+  else
+    return 25000 + (25000 * (1 << (desired_grade - 1)));
+}
+
+bool submersion_cost(struct char_data *ch, bool spend)
+{
+  int desired_grade = GET_GRADE(ch) + 1;
+  long karmacost = (desired_grade + 5) * 300;
+  long nuyencost = calculate_sub_nuyen_cost(desired_grade);
+
+  if (karmacost < 0) {
+    send_to_char("Congratulations, you've hit the ceiling! You can't increase submersion until the code is fixed to allow someone at your rank to do so.\r\n", ch);
+    mudlog_vfprintf(ch, LOG_SYSLOG, "lol init capped at %d for %s, karma overflowed... this game was not designed for that init level", GET_GRADE(ch), GET_CHAR_NAME(ch));
+    return FALSE;
+  }
+
+  long tke = 0;
+  if (karmacost > GET_KARMA(ch) || (GET_KARMA(ch) - karmacost) > GET_KARMA(ch)) {
+    send_to_char(ch, "You do not have enough karma to increase submersion. It will cost you %.2f karma.\r\n", ((float) karmacost / 100));
+    return FALSE;
+  }
+  if (nuyencost > GET_NUYEN(ch) || (GET_NUYEN(ch) - nuyencost) > GET_NUYEN(ch)) {
+    send_to_char(ch, "You do not have enough nuyen to increase submersion. It will cost you %ld nuyen.\r\n", nuyencost);
+    return FALSE;
+  }
+
+  switch (desired_grade) {
+    case 1:
+      tke = 0;
+      break;
+    case 2:
+      tke = 100;
+      break;
+    case 3:
+      tke = 200;
+      break;
+    default:
+      tke = 200 + ((GET_GRADE(ch)-2)*200);
+      break;
+  }
+  if (tke > GET_TKE(ch)) {
+    send_to_char(ch, "You do not have high enough TKE to initiate. You need %ld TKE.\r\n", tke);
```

Comment:
Is this function a copy/paste of the initiation cost code? If so, are the changes significant enough that code duplication is warranted, or can this be merged into the initiation cost code with a mode flag to switch between awakened and otaku initiations?